File: frameworks/domain.core/src/main/java/com/example/frameworks/domain/core/CollectionUtils.java
package com.example.frameworks.domain.core;

import lombok.*;

import javax.annotation.Nullable;
import java.util.*;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class CollectionUtils {

    public static <T> List<T> defaultList(@Nullable List<T> list) {
        return list != null ? list : Collections.emptyList();
    }

    public static <T> Boolean isEmpty(List<T> list) {
        return list == null || list.isEmpty();
    }

    public static <T> Boolean notEmpty(List<T> list) {
        return !isEmpty(list);
    }

    public static <T> Boolean isEmpty(Collection<T> coll) {
        return !isNotEmpty(coll);
    }

    public static <T> Boolean isNotEmpty(Collection<T> coll) {
        return coll != null && coll.size() > 0;
    }

}

File: frameworks/domain.core/src/main/kotlin/com/example/frameworks/domain/core/CollectionUtils.kt
@file:JvmName("CollectionUtils")

package com.example.frameworks.domain.core

fun <T> defaultList(list: List<T>?): List<T> {
    return list ?: emptyList()
}

fun <T> isEmpty(list: List<T>?): Boolean {
    return list == null || list.isEmpty()
}

fun <T> notEmpty(list: List<T>?): Boolean {
    return !isEmpty(list)
}

fun <T> isEmpty(coll: Collection<T>?): Boolean {
    return !isNotEmpty(coll)
}

fun <T> isNotEmpty(coll: Collection<T>?): Boolean {
    return coll != null && coll.isNotEmpty()
}
